_inc/lib/class.media.php

```diff
@@ -0,0 +1,505 @@
+		}
+
+		self::delete_media_history_file( $media_id, $original_file->file );
+		return delete_post_meta( $media_id, self::$WP_ORIGINAL_MEDIA );
+	}
+
+	/**
+	 * Clean `revision_history` of the given $media_id. it means:
+	 *   - remove all media files tied to the `revision_history` items.
+	 *   - clean `revision_history` meta data.
+	 *   - remove and clean the `original_media`
+	 * 
+	 * @param  number $media_id - media post ID
+	 * @return array results of removing these files
+	 */
+	public static function clean_revision_history( $media_id ) {
+		self::clean_original_media( $media_id );
+
+		$revision_history = self::get_revision_history( $media_id );
+		$total = count( $revision_history );
+		$updated_history = array();
+
+		if ( $total < 1 ) {
+			return $updated_history;
+		}
+
+		$updated_history = self::remove_items_from_revision_history(
+			$media_id,
+			array( 'from' => 0, 'to' => $total ),
+			$revision_history
+		);
+
+		return $updated_history;
+	}
+
+	/**
+	 * Edit media item process:
+	 *
+	 * - update attachment file
+	 * - preserve original media file
+	 * - trace revision history
+	 * 
+	 * @param  number $media_id - media post ID
+	 * @param  array $file_array - temporal file
+	 * @return {Post|WP_Error} Updated media item or a WP_Error is something went wrong.
+	 */
+	public static function edit_media_file( $media_id, $file_array ) {
+		$media_item = get_post( $media_id );
+		$has_original_media = self::get_original_media( $media_id );
+
+		if ( ! $has_original_media ) {
+			// The first time that the media is updated
+			// the original media is stored into the revision_history
+			$snapshot = self::get_snapshot( $media_item );
+			add_post_meta( $media_id, self::$WP_ORIGINAL_MEDIA, $snapshot, true );
+		}
+
+		// save temporary file in the correct location
+		$uploaded_file = self::save_temporary_file( $file_array, $media_id );
+
+		if ( is_wp_error( $uploaded_file ) ) {
+			self::remove_tmp_file( $file_array );
+			return $uploaded_file;
+		}
+
+		// revision_history control
+		self::register_revision( $media_item, $uploaded_file, $has_original_media );
+
+		$uploaded_path = $uploaded_file['file'];
+		$udpated_mime_type = $uploaded_file['type'];
+		$was_updated = update_attached_file( $media_id, $uploaded_path );
+
+		if ( ! $was_updated ) {
+			return WP_Error( 'update_error', 'Media update error' );
+		}
+
+		$new_metadata = wp_generate_attachment_metadata( $media_id, $uploaded_path );
+		wp_update_attachment_metadata( $media_id, $new_metadata );
+
+		// check maximum amount of revision_history
+		self::limit_revision_history( $media_id );
+
+		$edited_action = wp_update_post( (object) array(
+			'ID'              => $media_id,
+			'post_mime_type'  => $udpated_mime_type
+		), true );
+
+		if ( is_wp_error( $edited_action ) ) {
+			return $edited_action;
+		}
+
+		return $media_item;
+	}
+}
+
+// hook: clean revision history when the media item is deleted
+$clean_revision_history = function( $media_id ) {
```

Comment:
Unfortunately we need to support PHP 5.2.4 :(  

Anonymous functions were introduced in 5.3